rts/lra/coordinator/pom.xml

```diff
@@ -28,6 +28,11 @@
             <artifactId>microprofile-lra-api</artifactId>
             <version>${version.microprofile.lra.api}</version>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.fault-tolerance</groupId>
+            <artifactId>microprofile-fault-tolerance-api</artifactId>
```

Comment:
Is the compile scope considered as the right one for this?

Reply:
That's a good point, I cut and pasted the dependency from maven central. But you are right in querying this, I would have expected we'd need to be pulling in the smallrye implementation of FT. I will investigate further and update my PR depending upon what I find.

I did find that after I made the changes in this PR that I resolved the errors produced by the coordinator under heavy load (something like 5000 out of 5000 successful LRAs against two remote participants, it did take a couple of recovery passes for them all to finish okay). So I assumed my fix was good and did not drill down any further into the bulkhead.

Reply:
I think the API will be bundled to the final jar file and the quarkus/thorntail is then taking the smallrye implementation into consideration and start it.
The trouble could be with WFLY maybe. In fact when I was looking at the `pom.xml` I'm not sure, from this perspective, is good to have the compile scope for `microprofile-lra-api` either. The deployment should be using what's provided by app runtime. I hope we can consider this as a nuance that could be investigated when working on WFLY extension.

Reply:
Well I made some changes to how we consume the dependency although I didn't quite follow your last comment so you might want to check my latest push.

Reply:
My point is the best described in way that I think the dependency should be in scope `provided`. The app runtime then loads the api library. It won't be the application that defines it.

Reply:
+1